ramda.js

```diff
@@ -4705,6 +4707,42 @@
     R.gte = op(function _gte(a, b) { return a >= b; });
 
 
+    function createMaxMin(comparator, initialVal) {
+        return function(list) {
+            if (arguments.length === 0) {
+                throw noArgsException();
+            }
+            var idx = -1, winner = initialVal, computed;
+            while (++idx < list.length) {
+                computed = +list[idx];
+                if (comparator(computed, winner)) {
+                    winner = computed;
+                }
+            }
+            return winner;
+        };
+    }
+
+    function createMaxMinWith(comparator) {
+        return function(valueComputer, list) {
+            if (!(list && list.length > 0)) {
+                return;
```

Comment:
Hmm what if both `max` & `maxBy` had initial values of `-Infinity` and compared values were coerced to numbers that might solve most of the consistency gaps. Since the signature is arrays of numbers anything else could be outside the realm of documented support.